tests/lib/rules/array-bracket-newline.js

```diff
@@ -0,0 +1,526 @@
+                    type: "ArrayExpression",
+                    line: 5,
+                    column: 1
+                }
+            ]
+        },
+
+        // { multiline: true, minItems: 2 }
+        {
+            code: "var foo = [\n];",
+            options: [{ multiline: true, minItems: 2 }],
+            errors: [
+                {
+                    message: ERR_NO_BREAK_AFTER,
+                    type: "ArrayExpression",
+                    line: 1,
+                    column: 11
+                },
+                {
+                    message: ERR_NO_BREAK_BEFORE,
+                    type: "ArrayExpression",
+                    line: 2,
+                    column: 1
+                }
+            ]
+        },
+        {
+            code: "var foo = [\n1\n];",
+            options: [{ multiline: true, minItems: 2 }],
+            errors: [
+                {
+                    message: ERR_NO_BREAK_AFTER,
+                    type: "ArrayExpression",
+                    line: 1,
+                    column: 11
+                },
+                {
+                    message: ERR_NO_BREAK_BEFORE,
+                    type: "ArrayExpression",
+                    line: 3,
+                    column: 1
+                }
+            ]
+        },
+        {
+            code: "var foo = [1, 2];",
+            options: [{ multiline: true, minItems: 2 }],
+            errors: [
+                {
+                    message: ERR_BREAK_AFTER,
+                    type: "ArrayExpression",
+                    line: 1,
+                    column: 11
+                },
+                {
+                    message: ERR_BREAK_BEFORE,
+                    type: "ArrayExpression",
+                    line: 1,
+                    column: 16
+                }
+            ]
+        },
+        {
+            code: "var foo = [1,\n2];",
+            options: [{ multiline: true, minItems: 2 }],
+            errors: [
+                {
+                    message: ERR_BREAK_AFTER,
+                    type: "ArrayExpression",
+                    line: 1,
+                    column: 11
+                },
+                {
+                    message: ERR_BREAK_BEFORE,
+                    type: "ArrayExpression",
+                    line: 2,
+                    column: 2
+                }
+            ]
+        },
+        {
+            code: "var foo = [function foo() {\ndosomething();\n}];",
+            options: [{ multiline: true, minItems: 2 }],
+            errors: [
+                {
+                    message: ERR_BREAK_AFTER,
+                    type: "ArrayExpression",
+                    line: 1,
+                    column: 11
+                },
+                {
+                    message: ERR_BREAK_BEFORE,
+                    type: "ArrayExpression",
+                    line: 3,
+                    column: 2
+                }
+            ]
+        }
+    ]
+
```

Comment:
Could you add some tests cases where the array has comments:
- in-between the opening bracket and the first element
- the closing bracket and the last element
- between array elements
- just before the opening bracket
- just after the closing bracket (before the semicolon)
- any others you can think of :)